test/named-parameter-order-test.js

```diff
@@ -0,0 +1,36 @@
+var shuffle = require('shuffle-array');
+var test = require('tap').test;
+var docuri = require('..');
+
+var simulateWrongOrderObjectKeys = function(cb){
+  var orig = Object.keys;
+  Object.keys = function(obj){
+    var keys = orig(obj);
+    var real = keys.toString();
+    while (real === keys.toString())
+      shuffle(keys);
```

Comment:
I would not use shuffle in this case because it will cause the test to sporadically not fail. Just changing the order in a deterministic manner (eg switch first and last element) is enough and also frees us from an additional dependency.